### PR TITLE
Add version command

### PR DESCRIPTION
**Pull request by dansimau**
`astro version` will print the version of astro at the command line.
Version information will be set using ldflags at release time. For more
information, see:
https://stackoverflow.com/questions/11354518/golang-application-auto-build-versioning

Automated releases will be implemented in a separate PR (#14).

### CODE REVIEW

File: astro/cli/astro/cmd/cmds.go [hunk 1]
```diff
@@ -113,6 +113,16 @@ var planCmd = &cobra.Command{
 	},
 }
 
+var versionCmd = &cobra.Command{
+	Use:                   "version",
+	DisableFlagsInUseLine: true,
+	Short:                 "Print astro version",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fmt.Printf("astro version %s (%s) built %s\n", version, commit, date)
```
Comment (btromanova) [inline astro/cli/astro/cmd/cmds.go]: When I run version built locally, I'm getting 
```
astro version DEV (none) built unknown
```
Can we change this message so that it's clear what none and unknown stand for?
If it's hard to fill out the git hash and build time, you can skip this information all together or add tags, so that it'd be
```
astro version DEV (hash: unknown) built time: unknown
```

Comment (dansimau) [reply astro/cli/astro/cmd/cmds.go]: Sure, I think I'll just remove the hash and build time altogether.

Comment (dansimau) [reply astro/cli/astro/cmd/cmds.go]: OK, I updated this to address your comments.

Git hash and current date are now determined within Go at build time. This means the hash and date will be available even when compiling locally, e.g.:

```
astro version dev (8085049) built 2018-12-28T14:20:48+01:00
```

When the hash can't be determined:

```
astro version dev (hash unknown) built 2018-12-28T14:21:14+01:00
```

The releaser can update the version using linker flags:

```
dan-mbp-work:astro dan$ go build -o bin/astro -ldflags="-X github.com/uber/astro/astro/cli/astro/cmd.version=1.2.3" github.com/uber/astro/astro/cli/astro
dan-mbp-work:astro dan$ bin/astro version
astro version 1.2.3 (54691ae) built 2018-12-28T14:39:18+01:00
dan-mbp-work:astro dan$ 
```

That is addressed in #14 and will be retested after this change.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
-## 0.5.0 (UNRELEASED, 2018)
-
+## 0.5.0 (UNRELEASED, 2019)
+
+* Add `version` command
+* Propagate SIGINT and SIGTERM to terraform processes (#49)
+* Support detach flag with terraform 0.12 (#45)
+* Fix plan output for terraform 0.12 (#41)
+* Fix bug in initialization of allowed values (#43)
+* Don't pass varialbes to modules that don't delcare them (#40)
```
Comment (dansimau) [inline CHANGELOG.md]: Can we put this in a separate PR? They have nothing to do with this one.

(Also, typo: "variables")

File: astro/cli/astro/cmd/version.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (dansimau) [inline astro/cli/astro/cmd/version.go]: 2019

File: astro/tests/integration_test.go [hunk 4]
```diff
@@ -71,10 +71,18 @@ func stringVersionMatches(v string, versionConstraint string) bool {
 }
 
 // compiles the astro binary and returns the path to it.
-func compileAstro(dir string) (string, error) {
+func compileAstro(dir string, ldflags []string) (string, error) {
```
Comment (dansimau) [inline astro/tests/integration_test.go]: Let's just make this `goBuildFlags []string` and allow people to pass arbitrary flags.

File: astro/tests/integration_test.go [hunk 5]
```diff
@@ -219,3 +227,57 @@ func TestProjectPlanDetachSuccess(t *testing.T) {
 		})
 	}
 }
+
+func TestVersionDev(t *testing.T) {
+	dir, err := ioutil.TempDir("/tmp", "astro-tests")
+	if err != nil {
+		panic(err)
+	}
```
Comment (dansimau) [inline astro/tests/integration_test.go]: In tests just do:

```
require.NoError(t, err)
```

File: astro/tests/integration_test.go [hunk 6]
```diff
@@ -219,3 +227,57 @@ func TestProjectPlanDetachSuccess(t *testing.T) {
...
+	assert.Equal(t, "", result.Stderr.String())
+	assert.Equal(t, "astro version dev\n", result.Stdout.String())
+	assert.Equal(t, 0, result.ExitCode)
+}
+
+func TestVersionWithLdflags(t *testing.T) {
+	dir, err := ioutil.TempDir("/tmp", "astro-tests")
+	if err != nil {
+		panic(err)
+	}
```
Comment (dansimau) [inline astro/tests/integration_test.go]:
```
require.NoError(t, err)
```

File: astro/tests/integration_test.go [hunk 7]
```diff
@@ -219,3 +227,57 @@ func TestProjectPlanDetachSuccess(t *testing.T) {
...
+
+	err = command.Run()
+
+	require.NoError(t, err)
+
+	assert.Equal(t, "", stderrBytes.String())
+	assert.Equal(t, "astro version 1.2.3 (ab123) built 2019-01-01T10:00:00\n", stdoutBytes.String())
+}
+
+func TestPlanErorrsWithoutConfig(t *testing.T) {
```
Comment (dansimau) [inline astro/tests/integration_test.go]: Should we test `apply` too?

File: astro/tests/integration_test.go (hunk 5 above)
Comment (btromanova) [reply astro/tests/integration_test.go]: Done

File: astro/tests/integration_test.go (hunk 6 above)
Comment (btromanova) [reply astro/tests/integration_test.go]: Done, thanks

File: astro/tests/integration_test.go (hunk 7 above)
Comment (btromanova) [reply astro/tests/integration_test.go]: Added a test below.

File: astro/tests/integration_test.go [hunk 8]
```diff
@@ -272,12 +268,20 @@ func TestVersionWithLdflags(t *testing.T) {
 
 func TestPlanErorrsWithoutConfig(t *testing.T) {
 	dir, err := ioutil.TempDir("/tmp", "astro-tests")
-	if err != nil {
-		panic(err)
-	}
+	require.NoError(t, err)
 	defer os.RemoveAll(dir)
```
Comment (dansimau) [inline astro/tests/integration_test.go]: Defer should go one line above.

(You just gave me the same feedback 😄 )

File: astro/tests/integration_test.go [hunk 9]
```diff
@@ -272,12 +268,20 @@ func TestVersionWithLdflags(t *testing.T) {
...
 	result := RunTest(t, []string{"plan"}, "/tmp/astro-tests", "")
 	assert.Equal(t, "unable to find config file\n", result.Stderr.String())
 	assert.Equal(t, "", result.Stdout.String())
 	assert.Equal(t, 1, result.ExitCode)
 }
+
+func TestApplyErorrsWithoutConfig(t *testing.T) {
+	dir, err := ioutil.TempDir("/tmp", "astro-tests")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
```
Comment (dansimau) [inline astro/tests/integration_test.go]: Same here.

You could also make these two tests into a table test (if you wanted... 😉 )

File: astro/tests/integration_test.go (hunk 8 above)
Comment (btromanova) [reply astro/tests/integration_test.go]: Yep, changed it and combined tests.
